python/ql/lib/semmle/python/ApiGraphs.qll

```diff
@@ -339,6 +348,9 @@ module API {
       result = callee.getReturn() and
       result.getAnImmediateUse() = this
     }
+
+    /** Gets the number of arguments of this call. */
```

Comment:
Hmmm. That raises a good point.  
How do you want the `WithArity` filter to count the arity of a call?  

The current implementation is based on this `getNumArgument` predicate.  
I think `ruby` also counts the named parameters, so I'll change the implementation here to do that (instead of your suggestion).

Reply:
I'd say `WithArity` should count the positional parameters only, and (in another PR) we should add something like `WithArgument[name:]` for checking the presence of non-positional parameters.

Reply:
If this is for `WithArity`, I think the least surprising behavior would be to count both positional and keyword arguments. But I guess that actually depends on what a typical use case for `WithArity` would be, it's not something I've often needed myself in my QL writing. Do you have a canonical example of why this is useful?

Most of the time I would expect functions that take an optional amount of args as

```py3
def my_func(foo, bar=None, baz=None):
    # foo is required, bar and baz are not
    ...

my_func(1)
my_func(1, 2)
my_func(1, 2, 3)

my_func(1, baz=3) # <-- this line is interesting
my_func(1, bar=2)
```

So the interesting line highlighted would match `WithArity[2]` if you also include keyword arguments in the count, which is kinda funky, and maybe not what is the intended usage of `WithArity`... but I'm curious to see what the intended usage of `WithArity` actually is before making further judgement.

Traditional good style for Python code is to use a mix of positional and keyword arguments, using keyword arguments for "options" on a function -- such as `open('foo.txt', 'rw', encoding='utf-8')`

Reply:
> Do you have a canonical example of why this is useful? 

The `attr` method in JQuery. It is used to get/set attributes on DOM nodes.  
If you call it with 1 argument (e.g. `$("img").attr("src")`) the method will return the value of the attribute.  
If you call it with 2 arguments (e.g. `$("img").attr("src", "myimg.png")`) the method will set the new value of the attribute, and return the JQuery object so you can chain another call.   

The return value of the `attr` method is a source for some XSS queries, but only when it's called with 1 argument.   

I don't think those kinds of overloaded functions are common in Python.

Reply:
Oh, I never responded to this. I agree that these kind sof overloaded functions are not common in Python :+1:

based on the examples I made above, I suggest we go with only the positional arguments (and that my suggestion adjusting the QLDoc is accepted) -- that's the approach Asger suggested.

Reply:
> Oh, I never responded to this. I agree that these kind sof overloaded functions are not common in Python +1
> 
> based on the examples I made above, I suggest we go with only the positional arguments (and that my suggestion adjusting the QLDoc is accepted) -- that's the approach Asger suggested.

And I missed your response for a while.    
I've changed `getNumArgument` to only count positional arguments.